lexicon/client.py

```diff
@@ -10,6 +11,9 @@
     legacy_config_resolver, non_interactive_config_resolver,
 )
 
+TLDEXTRACT_CACHE_FILE_DEFAULT = os.path.join(os.path.dirname(__file__), '.tld_set')
```

Comment:
Two things:
* I propose to name the directory `lexicon_tld_set` to be clear about its usage
* I would like to avoid to clutter the Python installation path with cache if possible. Why not use `$(pwd)/.lexicon_tld_cache`, or `~/.lexicon_tld_cache` instead?

Reply:
Renaming to `.lexicon_tld_set` - 👍

Regarding the file location - I think that it should be predictable (i.e. fixed), so better to me is `~/.lexicon_tld_cache` instead of current directory. One reason to have this fixed is that the file is updated only when it is old (or missing), so it could be reused from multiple instances of `lexicon` running from different directories (but not concurrently, of course, because of risk of concurrent update).

There are some recommendations for application data storage (on Linux this is `/var/cache` or `~/.cache`, on Windows - ?, on iOS - ?), but I cannot find any library handling this, so easiest is to simply use `~/.lexicon_tld_cache` and let `expanduser` function place it in the correct home directory.

Alternative would be to always download the TLDs (not store it anywhere), but I prefer the above.

If you know about a library, which could select os-dependant path for cached data (i.e. data, which can be safely deleted by the user, but which is used by the application when it exists), it should be easy to change the default.